### PR TITLE
Master Project - Cleanup

### PR DESCRIPTION
**Pull request by Skgland**
* Remove unused imports
* Replace usage of deprecated functions with non-deprecated equivalents
* Fix rawtype and typesafety warnings
* replace println with logger 

# TODO
* [x] fix usagee of var
* [x] remove changes to automatically generated files in klighd.krendering

### CODE REVIEW

File: plugins/de.cau.cs.kieler.klighd.ui/src/de/cau/cs/kieler/klighd/ui/printing/dialog/RangeBlock.java [hunk 1]
```diff
@@ -106,36 +108,44 @@ public static Group createContents(final Composite parent, final DataBindingCont
...
-        final IObservableValue rangeFrom =
-                BeansObservables.observeValue(realm, options, PrintOptions.PROPERTY_RANGE_FROM);
-        bindings.bindValue(SWTObservables.observeText(textFrom, SWT.Modify), rangeFrom);
-        bindings.bindValue(SWTObservables.observeEnabled(textFrom), rangeValue);
+        final IObservableValue<Object> rangeFrom = 
+                BeanProperties.value(options.getClass().asSubclass(PrintOptions.class), PrintOptions.PROPERTY_RANGE_FROM).observe(realm, options);
+                // BeansObservables.observeValue(realm, options, PrintOptions.PROPERTY_RANGE_FROM);
+        var observedModifiedTextFrom = WidgetProperties.text(SWT.Modify).observe(textFrom); // SWTObservables.observeText(textFrom, SWT.Modify) 
+        bindings.bindValue(observedModifiedTextFrom, rangeFrom);
+        var observedEnabledFrom = WidgetProperties.enabled().observe(textFrom); //SWTObservables.observeEnabled(textFrom);
```
Comment (Skgland) [inline plugins/de.cau.cs.kieler.klighd.ui/src/de/cau/cs/kieler/klighd/ui/printing/dialog/RangeBlock.java:132]: fix usage of var

File: plugins/de.cau.cs.kieler.klighd.ui/src/de/cau/cs/kieler/klighd/ui/printing/dialog/RangeBlock.java [hunk 2]
```diff
@@ -106,36 +108,44 @@ public static Group createContents(final Composite parent, final DataBindingCont
...
-                BeansObservables.observeValue(realm, options, PrintOptions.PROPERTY_RANGE_TO);
-        bindings.bindValue(SWTObservables.observeText(textTo, SWT.Modify), rangeTo);
-        bindings.bindValue(SWTObservables.observeEnabled(textTo), rangeValue);
+        final IObservableValue<Object> rangeTo =
+                BeanProperties.value(options.getClass().asSubclass(PrintOptions.class), PrintOptions.PROPERTY_RANGE_TO).observe(realm, options);
+                // BeansObservables.observeValue(realm, options, PrintOptions.PROPERTY_RANGE_TO);
+        
+        var observedModifyTextTo = WidgetProperties.text(SWT.Modify).observe(textTo); //SWTObservables.observeText(textTo, SWT.Modify)
+        bindings.bindValue(observedModifyTextTo, rangeTo);
+        var observedEnabledTo = WidgetProperties.enabled().observe(textTo); //SWTObservables.observeEnabled(textTo)
```
Comment (Skgland) [inline plugins/de.cau.cs.kieler.klighd.ui/src/de/cau/cs/kieler/klighd/ui/printing/dialog/RangeBlock.java:147]: fix usage of var

File: plugins/de.cau.cs.kieler.klighd.ui/src/de/cau/cs/kieler/klighd/ui/printing/dialog/ScalingBlock.java [hunk 3]
```diff
@@ -183,23 +184,32 @@ public void widgetSelected(final SelectionEvent e) {
...
+                            PrintOptions.PROPERTY_PAGES_WIDE).observe(realm, dOptions); 
+            // BeansObservables.observeValue(realm, dOptions, PrintOptions.PROPERTY_PAGES_WIDE);
+            var observerWideSpinner = WidgetProperties.widgetSelection().observe(spinnerWide); // SWTObservables.observeSelection(spinnerWide);
+            bindings.bindValue(observerWideSpinner, pagesWide);
+
+            final IObservableValue<Object> pagesTall = 
+                    BeanProperties.value(dOptions.getClass().asSubclass(DiagramPrintOptions.class),
+                            PrintOptions.PROPERTY_PAGES_TALL).observe(realm, dOptions); 
+            // BeansObservables.observeValue(realm, dOptions, PrintOptions.PROPERTY_PAGES_TALL);
+            var observerTallSpinner = WidgetProperties.widgetSelection().observe(spinnerTall); // SWTObservables.observeSelection(spinnerTall);
```
Comment (Skgland) [inline plugins/de.cau.cs.kieler.klighd.ui/src/de/cau/cs/kieler/klighd/ui/printing/dialog/ScalingBlock.java:205]: fix usage of var

File: test/de.cau.cs.kieler.klighd.test/src/de/cau/cs/kieler/klighd/test/KGraphUtilTest.java [hunk 4]
```diff
@@ -40,16 +41,18 @@
 
     @Test
     public void getConnectedEdgesTest01() {
-        final Iterator<KEdge> res = KGraphUtil.getConnectedEdges(
-                getLast(getLast(getLast(testModel.getChildren()).getChildren()).getOutgoingEdges()));
+        var edge =
```
Comment (Skgland) [inline test/de.cau.cs.kieler.klighd.test/src/de/cau/cs/kieler/klighd/test/KGraphUtilTest.java:44]: fix usage of var

File: test/de.cau.cs.kieler.klighd.test/src/de/cau/cs/kieler/klighd/test/KGraphUtilTest.java [hunk 5]
```diff
@@ -40,16 +41,18 @@
...
 
-        Assert.assertThat(res, IsIteratorWithSize.iteratorWithSize(3));
+        MatcherAssert.assertThat("", res, IsIteratorWithSize.iteratorWithSize(3));
     }
 
     @Test
     public void getConnectedEdgesTest01a() {
-        final Iterator<KEdge> edges = KGraphUtil.getConnectedEdges(
-                getLast(getLast(getLast(testModel.getChildren()).getChildren()).getOutgoingEdges()));
+        var edge = getLast(
```
Comment (Skgland) [inline test/de.cau.cs.kieler.klighd.test/src/de/cau/cs/kieler/klighd/test/KGraphUtilTest.java:53]: fix usage of var

File: plugins/de.cau.cs.kieler.klighd.ui/src/de/cau/cs/kieler/klighd/ui/printing/dialog/PrinterBlock.java [hunk 6]
```diff
@@ -95,21 +95,28 @@ public static Group createContents(final Composite parent, final DataBindingCont
...
+                */
                 if (data.printToFile) {
                     return data.fileName;
                 } else {
                     return data.name;
                 }
             }
         };
-        bindings.bindValue(SWTObservables.observeText(printerLabel), computedValue);
+        final var labelText = WidgetProperties.text().observe(printerLabel); // SWTObservables.observeText(printerLabel);
```
Comment (Skgland) [inline plugins/de.cau.cs.kieler.klighd.ui/src/de/cau/cs/kieler/klighd/ui/printing/dialog/PrinterBlock.java:118]: fix usage of var

File: plugins/de.cau.cs.kieler.kgraph.text.ui/src/de/cau/cs/kieler/kgraph/text/ui/contentassist/KGraphProposalProvider.java [hunk 7]
```diff
@@ -471,13 +472,19 @@ private void valueProposal(final ContentAssistContext context,
                         break;
                     case OBJECT:
                         try {
+                        
                             proposal = "\""
-                                    + optionData.getOptionClass().newInstance().toString()
+                                    + optionData.getOptionClass().getConstructor().newInstance().toString()
```
Comment (NiklasRentzCAU) [inline plugins/de.cau.cs.kieler.kgraph.text.ui/src/de/cau/cs/kieler/kgraph/text/ui/contentassist/KGraphProposalProvider.java:477]: The deprecation warning of .newInstance() suggests to use .getDeclaredConstructor().newInstance(), I would suggest to follow that advice.

File: plugins/de.cau.cs.kieler.kgraph.text.ui/src/de/cau/cs/kieler/kgraph/text/ui/contentassist/KGraphProposalProvider.java [hunk 8]
```diff
@@ -471,13 +472,19 @@ private void valueProposal(final ContentAssistContext context,
...
                                     + "\"";
+                        
                         } catch (final InstantiationException e) {
                             proposal = "\"\"";
                         } catch (final IllegalAccessException e) {
                             proposal = "\"\"";
+                        }catch (IllegalArgumentException | InvocationTargetException
+                                | NoSuchMethodException | SecurityException e) {
+                            proposal = "\"\"";
+                            e.printStackTrace();
```
Comment (NiklasRentzCAU) [inline plugins/de.cau.cs.kieler.kgraph.text.ui/src/de/cau/cs/kieler/kgraph/text/ui/contentassist/KGraphProposalProvider.java:487]: Why does this exception have a different handling than the others?

File: plugins/de.cau.cs.kieler.kgraph.text.ui/src/de/cau/cs/kieler/kgraph/text/ui/contentassist/KGraphProposalProvider.java [hunk 9]
```diff
@@ -471,13 +472,19 @@ private void valueProposal(final ContentAssistContext context,
...
                             proposal = "\""
-                                    + optionData.getOptionClass().newInstance().toString()
+                                    + optionData.getOptionClass().getConstructor().newInstance().toString()
                                     + "\"";
+                        
                         } catch (final InstantiationException e) {
                             proposal = "\"\"";
                         } catch (final IllegalAccessException e) {
                             proposal = "\"\"";
+                        }catch (IllegalArgumentException | InvocationTargetException
```
Comment (NiklasRentzCAU) [inline plugins/de.cau.cs.kieler.kgraph.text.ui/src/de/cau/cs/kieler/kgraph/text/ui/contentassist/KGraphProposalProvider.java:484]: Missing whitespace after }

File: plugins/de.cau.cs.kieler.klighd.ui.view/src/de/cau/cs/kieler/klighd/ui/view/controller/ViewUpdateControllerFactory.java [hunk 10]
```diff
@@ -320,7 +320,7 @@ public static AbstractViewUpdateController getNewInstance(final String controlle
                     mucm.idControllerMapping.get(controllerID);
             if (clazz != null) {
                 try {
-                    AbstractViewUpdateController newInstance = clazz.newInstance();
+                    AbstractViewUpdateController newInstance = clazz.getConstructor().newInstance();
```
Comment (NiklasRentzCAU) [inline plugins/de.cau.cs.kieler.klighd.ui.view/src/de/cau/cs/kieler/klighd/ui/view/controller/ViewUpdateControllerFactory.java:323]: Same for all reflective new instances: the deprecation warning suggests to use getDeclaredConstructor()

File: plugins/de.cau.cs.kieler.klighd.ui/src/de/cau/cs/kieler/klighd/ui/printing/dialog/RangeBlock.java [hunk 11]
```diff
@@ -106,36 +109,46 @@ public static Group createContents(final Composite parent, final DataBindingCont
         // the update of 'rangeValue' is than caused by the de-selection of 'allValue' leading to
         //  a re-computation of rangValue by means of the above mentioned listeners (crazy shit ...);
         //  the than calculated result is used below for enabling the "from" & "to" text fields.
-        final IObservableValue rangeValue = new ComputedValue(realm) {
+        final IObservableValue rangeValue = new ComputedValue<>(realm) {
```
Comment (NiklasRentzCAU) [inline plugins/de.cau.cs.kieler.klighd.ui/src/de/cau/cs/kieler/klighd/ui/printing/dialog/RangeBlock.java]: For me this replaces the warning with an error: "'<>' cannot be used with anonymous classes"

File: plugins/de.cau.cs.kieler.klighd.ui/src/de/cau/cs/kieler/klighd/ui/printing/dialog/PrinterBlock.java [hunk 12]
```diff
@@ -95,21 +95,29 @@ public static Group createContents(final Composite parent, final DataBindingCont
         final Realm realm = bindings.getValidationRealm();
 
         // Observable to display the printer resp. file name, if print to file is set.
-        final ComputedValue computedValue = new ComputedValue(realm) {
+        final ComputedValue<?> computedValue = new ComputedValue<>(realm) {
```
Comment (NiklasRentzCAU) [inline plugins/de.cau.cs.kieler.klighd.ui/src/de/cau/cs/kieler/klighd/ui/printing/dialog/PrinterBlock.java:99]: For me this replaces the warning with an error: "'<>' cannot be used with anonymous classes"

File: test/de.cau.cs.kieler.klighd.piccolo.test/src/de/cau/cs/kieler/klighd/piccolo/test/DiagramClipGlobalPortLabelVisiblityConfigTest.java [hunk 13]
```diff
@@ -279,13 +280,13 @@ public void test() throws InterruptedException {
                 waitAmoment();
 
                 moveTo((int) (port0pos.x), (int) (port0pos.y));
-                Assert.assertThat(Pair.of(viewer.getControl(), port0pos), portMatcher);
+                MatcherAssert.assertThat("",Pair.of(viewer.getControl(), port0pos), portMatcher);
```
Comment (NiklasRentzCAU) [inline test/de.cau.cs.kieler.klighd.piccolo.test/src/de/cau/cs/kieler/klighd/piccolo/test/DiagramClipGlobalPortLabelVisiblityConfigTest.java:283]: missing whitespace after first parameter (3 times)

File: test/de.cau.cs.kieler.klighd.piccolo.test/src/de/cau/cs/kieler/klighd/piccolo/test/DiagramClipPortLabelVisibilityTest.java [hunk 14]
```diff
@@ -279,13 +280,13 @@ public void test() throws InterruptedException {
                 waitAmoment();
 
                 moveTo((int) (port0pos.x), (int) (port0pos.y));
-                Assert.assertThat(Pair.of(viewer.getControl(), port0pos), portMatcher);
+                MatcherAssert.assertThat("",Pair.of(viewer.getControl(), port0pos), portMatcher);
```
Comment (NiklasRentzCAU) [inline test/de.cau.cs.kieler.klighd.piccolo.test/src/de/cau/cs/kieler/klighd/piccolo/test/DiagramClipPortLabelVisibilityTest.java:283]: also missing whitespace

File: test/de.cau.cs.kieler.klighd.piccolo.test/src/de/cau/cs/kieler/klighd/piccolo/test/DiagramClipTest.java [hunk 15]
```diff
@@ -200,12 +201,12 @@ public void test() throws InterruptedException {
             waitAmoment();
 
             moveTo((int) (port0pos.x), (int) (port0pos.y));
-            Assert.assertThat(Pair.of(viewer.getControl(), port0pos), IS_BLACK);
+            MatcherAssert.assertThat("",Pair.of(viewer.getControl(), port0pos), IS_BLACK);
```
Comment (NiklasRentzCAU) [inline test/de.cau.cs.kieler.klighd.piccolo.test/src/de/cau/cs/kieler/klighd/piccolo/test/DiagramClipTest.java:204]: also missing whitespace

File: test/de.cau.cs.kieler.klighd.piccolo.test/src/de/cau/cs/kieler/klighd/piccolo/test/highlightedEdgeToForeground/HighlightedEdgeToForegroundTest.java [hunk 16]
```diff
@@ -89,58 +89,64 @@ public void configure(final Binder binder) {
         new ContextViewer(shell).setModel(viewContext, true);
 
         viewContext.update(null);
-        new LightDiagramLayoutConfig(viewContext)
-            .animate(false)
-            .zoomStyle(ZoomStyle.ZOOM_TO_ACTUAL_SIZE)
-            .performLayout();
+        new LightDiagramLayoutConfig(viewContext).animate(false)
+                .zoomStyle(ZoomStyle.ZOOM_TO_ACTUAL_SIZE).performLayout();
```
Comment (NiklasRentzCAU) [inline test/de.cau.cs.kieler.klighd.piccolo.test/src/de/cau/cs/kieler/klighd/piccolo/test/highlightedEdgeToForeground/HighlightedEdgeToForegroundTest.java:93]: Where is this better readable than the previous version? I would leave it as it was before.

File: test/de.cau.cs.kieler.klighd.piccolo.test/src/de/cau/cs/kieler/klighd/piccolo/test/highlightedEdgeToForeground/HighlightedEdgeToForegroundTest.java [hunk 17]
```diff
@@ -89,58 +89,64 @@ public void configure(final Binder binder) {
...
+        // final KShapeLayout firstChildNodeLayout =
+        // viewContext.getViewModel().getChildren().get(0).getChildren().get(0);
+        // final int firstChildNodeYPos = Math.round(firstChildNodeLayout.getYpos());
+        // final int firstClickXPos = 5 + 100;
+        //
+        // clickOn(firstClickXPos, firstChildNodeYPos);
+        // waitAmoment();
+        //
+        // clickOn(4, 4);
+        // waitAmoment();
```
Comment (NiklasRentzCAU) [inline test/de.cau.cs.kieler.klighd.piccolo.test/src/de/cau/cs/kieler/klighd/piccolo/test/highlightedEdgeToForeground/HighlightedEdgeToForegroundTest.java:142]: This change seems unnecessary, especially as it destroys the indentation in the commented out code

File: plugins/de.cau.cs.kieler.klighd.piccolo.batik/.classpath [hunk 18]
```diff
@@ -11,16 +11,12 @@
 	<classpathentry exported="true" kind="lib" path="lib/lib/js.jar"/>
 	<classpathentry exported="true" kind="lib" path="lib/lib/pdf-transcoder.jar"/>
 	<classpathentry exported="true" kind="lib" path="lib/lib/xalan-2.6.0.jar"/>
-	<classpathentry exported="true" kind="lib" path="lib/lib/xerces_2_5_0.jar"/>
```
Comment (NiklasRentzCAU) [inline plugins/de.cau.cs.kieler.klighd.piccolo.batik/.classpath:14]: I am not sure if the changes from this commit removes anything needed for this piccolo.batik plugin to work properly. Have you tested this export feature before removing  these dependencies?

**Review by NiklasRentzCAU — CHANGES_REQUESTED** [review]
First of all: thank you for tackling to resolve some deprecation warnings and issues in this project. As this PR contains quite some different aspects, I have some comments for them:
First of all, some of the reformatting of comment texts and source code seem to originate from some automatic optimization: As commented on some parts, these changes seem unnecessary and even make parts of the code less readable than they were before.
Second, the removed dependenies from the piccolo.batik bundle could be disruptive: did you test if exporting via the batik export still works as before? I did not work in that part of the code yet, so I don't quite know yet why these dependencies were there in the first place, but I would guess they are needed.
I also left some more comments, please review them before we continue with this PR.

File: plugins/de.cau.cs.kieler.kgraph.text.ui/src/de/cau/cs/kieler/kgraph/text/ui/contentassist/KGraphProposalProvider.java (hunk 8 above)
Comment (Skgland) [reply plugins/de.cau.cs.kieler.kgraph.text.ui/src/de/cau/cs/kieler/kgraph/text/ui/contentassist/KGraphProposalProvider.java:487]: The main reason is that the InvocationTargetException was previously a different  RuntimeException and therefore not caught. 
The NoSuchMethodeExcpetion would previously fall under the InstantiationException and should probably be handled as such.
The IllegalArgumentException should not be possible as we request a nullary Constructor and then call the constructor without arguments, as such it shouldn't make a difference. 
I don't know why I am explicitly handling the SecurityException.

Comment (Skgland) [reply plugins/de.cau.cs.kieler.kgraph.text.ui/src/de/cau/cs/kieler/kgraph/text/ui/contentassist/KGraphProposalProvider.java:487]: As discussed only the InvocationTargetException is handled separately now

**Comment by Skgland** [comment]
All remarks should be resolved.
I removed the commit removing the batik dependencies and opened the issue #97, regarding the build problems in newer java versions

File: plugins/de.cau.cs.kieler.klighd.ui/src/de/cau/cs/kieler/klighd/ui/printing/dialog/RangeBlock.java (hunk 11 above)
Comment (NiklasRentzCAU) [reply plugins/de.cau.cs.kieler.klighd.ui/src/de/cau/cs/kieler/klighd/ui/printing/dialog/RangeBlock.java]: Your fix did not fix this error for me, replacing the <> with <Boolean> as well fixes it for me.

**Comment by NiklasRentzCAU** [comment]
With that one more comment fixed this can soon be merged.

That comment from me the last time still holds though: "Some of the reformatting of comment texts and source code seem to originate from some automatic optimization: As commented on some parts, these changes seem unnecessary and even make parts of the code less readable than they were before."
With that I especially mean some of the changes where previously neatly arranged code is changed around just to justify the automatic code layouting, for example [here](https://github.com/kieler/KLighD/pull/91/files#diff-ccf6234e8a17a819a89a219364e51c78cc0cb2d0343eb31748ceac991a94b76eR109-R116), [here](https://github.com/kieler/KLighD/pull/91/files#diff-ccf6234e8a17a819a89a219364e51c78cc0cb2d0343eb31748ceac991a94b76eR258-R262) and a few other cases as well. I would rather revoke these "optimizations".

File: plugins/de.cau.cs.kieler.klighd.ui/src/de/cau/cs/kieler/klighd/ui/printing/dialog/RangeBlock.java (hunk 11 above)
Comment (Skgland) [reply plugins/de.cau.cs.kieler.klighd.ui/src/de/cau/cs/kieler/klighd/ui/printing/dialog/RangeBlock.java]: Oh, I thought I had fixed this but apparently I somehow missed this in 145ba1a33fc32914952fc95d94f95a533f2871f5

Comment (Skgland) [reply plugins/de.cau.cs.kieler.klighd.ui/src/de/cau/cs/kieler/klighd/ui/printing/dialog/RangeBlock.java]: This should now be fixed in 75cc039e7814e12333012c092d4401be538072ad

File: plugins/de.cau.cs.kieler.klighd.ui/src/de/cau/cs/kieler/klighd/ui/printing/dialog/PrinterBlock.java (hunk 12 above)
Comment (Skgland) [reply plugins/de.cau.cs.kieler.klighd.ui/src/de/cau/cs/kieler/klighd/ui/printing/dialog/PrinterBlock.java]: fixed in 145ba1a33fc32914952fc95d94f95a533f2871f5

**Comment by Skgland** [comment]
@NiklasRentzCAU Is the formatting now acceptable?

**Comment by NiklasRentzCAU** [comment]
Yes, the formatting looks fine now. But I noticed that replacing the `SWTObservables` deprecated in newer versions with `WidgetProperties` is only available since Java 9, whereas this project still uses Java 8 LTS for its build and requirements, making this change yet incompatible before dropping support for Java 8 and moving to newer LTS such as Java 11.

**Comment by Skgland** [comment]
> Yes, the formatting looks fine now. But I noticed that replacing the `SWTObservables` deprecated in newer versions with `WidgetProperties` is only available since Java 9, whereas this project still uses Java 8 LTS for its build and requirements, making this change yet incompatible before dropping support for Java 8 and moving to newer LTS such as Java 11.

@NiklasRentzCAU I have pushed a commit replacing `org.eclipse.jface.databinding.swt.typed.WidgetProperties` with `org.eclipse.jface.databinding.swt.typed.WidgetProperties` as the later should not the problem you mentioned.

**Review by NiklasRentzCAU — APPROVED** [review]
No more objections, thanks for contributing these changes!

**Pull request merged**